plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -50,51 +52,31 @@ def project
     @project ||= kuber_release.project
   end
 
-  def watch_deployment
-    # Doing this for now to force the loading of these constants.
-    # I was getting a circular dependency error otherwise
-    # TODO: hunt down this issue and fix it properly
-    Watchers::PodWatcher ; Watchers::ReplicationControllerWatcher ; Watchers::EventWatcher
-
-    Thread.new do
-      kuber_release.watch_pods(&method(:handle_pod_notice))
-    end
-
-    # Thread.new { sleep(5) ; kuber_release.watch_pod_events }
-  end
-
   private
 
-  def handle_pod_notice(notice, watcher)
-    release_doc = kuber_release.docs_by_role[notice.role]
-
-    if release_doc.nil?
-      Kubernetes::Util.log "ERROR: could not find role", role: notice.role, available_roles: kuber_release.docs_by_role.keys
-      return
-    end
-
-    ready_count = watcher.num_ready
-    if ready_count > release_doc.replicas_live
-      release_doc.update_replica_count(ready_count)
-      release_doc.save!
-      update_release_status
-      log 'Another replica is live', role: notice.role, count: ready_count, r_status: kuber_release.status, r_doc_status: release_doc.status
-    end
-  end
-
-  def update_release_status
-    if kuber_release.release_docs.all?(&:live?)
-      deploy_is_live
-    elsif kuber_release.release_docs.any?(&:spinning_up?)
-      kuber_release.update_attribute(:status, 'spinning_up') unless kuber_release.spinning_up?
+  # TODO: Remove this dummy data when proper watchers created.
+  def publish_fake_updates
+    Thread.new do
+      @kuber_release.release_docs.each do |release_doc|
+        release_doc.replica_target.times do
+          sleep 2
+          Celluloid::Notifications.publish("#{release_doc.replication_controller_name}", fake_event)
```

Comment:
Good point, there is a gem that's supposed to autoload everything, but I moved the require into the plugin engine init file.